src/edu/csus/ecs/pc2/imports/ccs/ICPCTSVLoader.java

```diff
@@ -319,17 +329,57 @@ public static Group[] loadGroups(String filename) throws Exception {
             } catch (NumberFormatException e) {
                 throw new InvalidValueException("Expecting group number got '" + numberStr + "' on line " + lines[i], e);
             }
-
-            Group group = new Group(groupName);
+            
+            Group group = findGroupById(authoritativeGroups, number);
+            if (group == null) {
+                group = new Group(groupName);
+            }
+            
+            group.setDisplayName(groupName);
             group.setGroupId(number);
-
-            groups[groupAccount] = group;
-            groupAccount++;
+            
+            groupList.add(group);
         }
-
+        
+        /**
+         * Add groups from authoritativeGroups if missing from groupList.
+         */
```

Comment:
I do not understand the purpose of, or the need for, the following for-loop.  The comment says that it's to add to `groupList` any groups that were in `authoritativeGroups` but are missing from `groupList`.  However, the new code added to the previous for-loop (just above here) obtains a group from the input file, then looks to see if that group is in `authoritativeGroups` (using `Group group = findGroupById(authoritativeGroups, number);`).  It then adds the group (either the one found in `authoritativeGroups` or else a newly-created group) to `groupList`.   Therefore, at THIS point in the code _every_ group found in `authoritativeGroups` MUST also now be in `groupList` -- so what's the point in the entire for-loop following this comment?
Or am I missing something?

Reply:
BTW, I would not block approval for this -- but I don't like having chunks of code that are useless.   (Unless I'm wrong about what's going on, in which case please explain how I'm wrong...)

Reply:
The comment is correct.     Merge may imply that the number of groups in the input file are identical to
the number/data for the existing authoritativeList.   That would be a bug.   There may be more or different
groups in the authoritative list than there are in the input file.    The merge is a union of groups in
the input file and the authoritativeList (model).   For example there could be 12 authoritativeList
groups and only 4 groups in the input file.  There should be a minimum of 12 groups output.  Without
the second loop there would be 4 groups output.

Logic is:
The top loop merges the authoritative/existing group data and the groups from the input file.  The comment is:
   "Add groups from authoritativeGroups if missing from groupList."
The bottom loop adds any missing groups from the authoritative list into the method's output groupList

Reply:
Ok, it took me a while but now I get why the second loop is needed.  However, it took me a LONG time to understand what the second loop is actually doing, due to very bad naming (see below).  I believe the code works -- but it's really confusing to the reader (again, see below).